x-pack/plugins/cloud_security_posture/public/pages/benchmarks/benchmarks_table.tsx

```diff
@@ -5,60 +5,138 @@
  * 2.0.
  */
 
-import { EuiBasicTable, type EuiBasicTableColumn } from '@elastic/eui';
+import {
+  EuiLink,
+  EuiBasicTable,
+  type EuiBasicTableColumn,
+  type EuiBasicTableProps,
+  type Pagination,
+  type CriteriaWithPagination,
+} from '@elastic/eui';
 import React from 'react';
 import moment from 'moment';
+import { Link, useHistory } from 'react-router-dom';
 import { TABLE_COLUMN_HEADERS } from './translations';
-import type { CspBenchmarkIntegration } from './types';
+import type { Benchmark } from '../../../common/types';
+import { pagePathGetters } from '../../../../fleet/public';
+import { useKibana } from '../../common/hooks/use_kibana';
 
-interface BenchmarksTableProps {
-  benchmarks: CspBenchmarkIntegration[];
+interface BenchmarksTableProps
+  extends Pick<EuiBasicTableProps<Benchmark>, 'loading' | 'error' | 'noItemsMessage'>,
+    Pagination {
+  benchmarks: Benchmark[];
+  setQuery(pagination: CriteriaWithPagination<Benchmark>): void;
   'data-test-subj'?: string;
 }
 
-const BENCHMARKS_TABLE_COLUMNS: Array<EuiBasicTableColumn<CspBenchmarkIntegration>> = [
+const AgentPolicyButtonLink = ({ name, id: policyId }: { name: string; id: string }) => {
+  const { http } = useKibana().services;
+  return (
+    <EuiLink
+      href={http.basePath.prepend(pagePathGetters.policy_details({ policyId }).join(''))}
+      title={name}
+    >
+      {name}
+    </EuiLink>
+  );
+};
+
+const BENCHMARKS_TABLE_COLUMNS: Array<EuiBasicTableColumn<Benchmark>> = [
   {
-    field: 'integration_name',
-    name: TABLE_COLUMN_HEADERS.INTEGRATION_NAME,
-    dataType: 'string',
+    field: 'package_policy.name',
+    name: TABLE_COLUMN_HEADERS.BENCHMARK_RULES,
+    render: (v, r) => (
+      <Link to={`/benchmark/${r.package_policy.id}/${r.package_policy.policy_id}/rules`} title={v}>
+        {v}
+      </Link>
```

Comment:
this links is kinda fragile, as it's corresponding path is at `allNavigationItems.rules.path` and equals to `/benchmark/:packageId/:policyId/rules` 

so we can't use the path as-is, we need to interpolate and replace the`:param` with its value. 

there's a nice way to do this type safe way, using some tricks i found on the web:

1. extract params from string
```typescript
// https://twitter.com/danvdk/status/1301707026507198464?lang=en

type Params<T extends string> = string extends T
  ? Record<string, string>
  : T extends `${infer Start}:${infer Param}/${infer Rest}`
  ? { [k in Param | keyof Params<Rest>]: string }
  : T extends `${infer Start}:${infer Param}`
  ? { [k in Param]: string }
  : {};

```

2 - add `link` fn to make type-safe links

```typescript
const captureVariables = /:(\w*)/gi;

function link<T extends string>(
  ...args: {} extends Params<T> ? [path: T] : [path: T, params: Params<T>]
) {
  const [path, params] = args;
  if (!params) return path;

  return path.replace(captureVariables, (_, variable) =>
    encodeURIComponent(params[variable as keyof typeof params] as string)
  );
}
```

now we can do:
- `link(allNavigationItems.benchmark.path)` =  `/benchmark`
- `link(allNavigationItems.rules.path, { packageId: "1", policyId: "1")`  = `/benchmark/1/1/rules`

and both will be type safe

IMO this is a nice solution and personally i'd go with it to tackle this. thoughts?

Reply:
IMO this is very complex and I don't want us to maintain it. I think we should use [`generatePath`](https://v5.reactrouter.com/web/api/generatePath) which is not super type-safe, but a good compromise IMO, and might be more type-safe in the future

Reply:
true that is it is rather complex 😅 

but the current generatePath isn't type safe, so we could have errors there. 

btw - next react-router types [version](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50295/files#diff-a64dab61f590d4d653f05bd16e2005ea54eeba8e6e943f041bb2af2f3218655cR137-R145) uses the same type trickery and fixes this

Reply:
changed to it anyway, does look much cleaner regardless of types